django/applications/catmaid/control/treenodeexport.py

```diff
@@ -108,6 +108,8 @@ def create_path(self, treenode):
         needed multiple times, will be cached. This function will also make
         sure the path exists and is ready to be written to.
         """
+        # XXX This almost always returns a string; the sole exception is
```

Comment:
Might be worth reviewing this logic - might lead to weird behaviour in rare cases

Reply:
That's true and the exception handling logic is not intuitive. Something like this would be better I believe:
```
from django.core.exceptions import ImproperlyConfigured
…
# Everything is fine if the path exists and is writable
if not os.path.exists(treenode_path):
   os.makedirs(treenode_path)
if not os.access(treenode_path, os.W_OK):
    raise ImproperlyConfigured("Treenode export path is not writable")
return treenode_path
```

Reply:
Hoping with the new revision it's what you're looking for